### PR TITLE
[Admin][TerritorialCouncil] Add Political committee members on Territorial council admin pages

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

**Comment by illusionOfParadise** [comment]
[ch12237]
[ch12246]

File: translations/messages.fr.yml [hunk 1]
```diff
@@ -918,6 +918,23 @@ territorial_council_membership_log.is_not_resolved: Ce log n'est pas encore rés
...
+political_committee.membership.quality.regional_council_president: Président(e) conseil régional
+political_committee.membership.quality.departmental_council_president: Président(e) conseil départemental
+political_committee.membership.quality.mayor: Maire
+political_committee.membership.quality.leader: Chef de file
+political_committee.membership.quality.regional_councilor: Conseiller(ère) régional(e) désigné(e)
+political_committee.membership.quality.department_councilor: Conseiller(ère) départemental(e) désigné(e)
+political_committee.membership.quality.city_councilor: Conseiller(ère) municipal(e) désigné(e)
+political_committee.membership.quality.consular_conselor: Conseiller(ère) consulaire désigné(e)
+political_committee.membership.quality.committee_supervisor: Animateur(trice) local(e) désigné(e)
+political_committee.membership.quality.elected_candidate_adherent: Adhérent(e) désigné(e) désigné(e)
```
Comment (kevin-verschaeve) [inline translations/messages.fr.yml:937]:
```suggestion
political_committee.membership.quality.elected_candidate_adherent: Adhérent(e) désigné(e)
```

File: src/Admin/TerritorialCouncil/TerritorialCouncilMembershipAdmin.php [hunk 2]
```diff
@@ -80,6 +112,35 @@ protected function configureDatagridFilters(DatagridMapper $filter)
...
+                    if (null !== $value['value']) {
+                        if (!\in_array('adherent', $qb->getAllAliases())) {
+                            $qb
+                                ->leftJoin("$alias.adherent", 'adherent')
+                                ->leftJoin('adherent.politicalCommitteeMembership', 'pcMembership')
+                            ;
+                        }
+
+                        $condition = true === $value['value'] ? 'NOT' : '';
+                        $qb->andWhere("pcMembership.id IS $condition NULL");
```
Comment (kevin-verschaeve) [inline src/Admin/TerritorialCouncil/TerritorialCouncilMembershipAdmin.php:136]: this should be into the if above non ? if we don't enter the if, pcMembership is not defined

File: src/Admin/TerritorialCouncil/TerritorialCouncilMembershipAdmin.php [hunk 3]
```diff
@@ -67,6 +68,37 @@ protected function configureDatagridFilters(DatagridMapper $filter)
...
+                    }
+
+                    if (!\in_array('adherent', $qb->getAllAliases())) {
+                        $qb
+                            ->leftJoin("$alias.adherent", 'adherent')
+                            ->leftJoin('adherent.politicalCommitteeMembership', 'pcMembership')
+                        ;
+                    }
+                    $qb
+                        ->leftJoin('pcMembership.qualities', 'pcQuality')
```
Comment (kevin-verschaeve) [inline src/Admin/TerritorialCouncil/TerritorialCouncilMembershipAdmin.php:94]: this should be into the if above non ? if we don't enter the if, pcMembership seems not defined

File: src/Admin/TerritorialCouncil/TerritorialCouncilMembershipAdmin.php (hunk 2 above)
Comment (illusionOfParadise) [reply src/Admin/TerritorialCouncil/TerritorialCouncilMembershipAdmin.php:136]: No, the `if` above exists to `adherent` and `pcMembership` aliases, if they have not yet been added.

File: src/Admin/TerritorialCouncil/TerritorialCouncilMembershipAdmin.php [hunk 4]
```diff
@@ -80,6 +112,35 @@ protected function configureDatagridFilters(DatagridMapper $filter)
...
+                                ->leftJoin("$alias.adherent", 'adherent')
+                                ->leftJoin('adherent.politicalCommitteeMembership', 'pcMembership')
+                            ;
+                        }
+
+                        $condition = true === $value['value'] ? 'NOT' : '';
+                        $qb->andWhere("pcMembership.id IS $condition NULL");
+
+                        return true;
+                    } else {
```
Comment (ottaviano) [inline src/Admin/TerritorialCouncil/TerritorialCouncilMembershipAdmin.php:139]: we can make this IF more readable with:
```php
if (a) {
  /**
   * some lines of code
   */
    return true;
} else {
    return false;
}
```

=> 
```php
if (!a) {
    return false;
}

/**
 * some lines of code
 */
return true;

```

**Review by ottaviano — APPROVED** [review]

**Pull request merged**